### PR TITLE
session: initialize session bind handler when creating session according to `createSessionWithOpt `

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve?

Problem Summary: 

When running evolve tasks, the `sessionContext` used by the background goroutine is not initialized with `SessionBindHandler`, which causes evolve task failures.

### What is changed and how it works?
Initialize session bind handler when creating session according to `createSessionWithOpt`.

### CODE REVIEW

**Comment by qw4990** [comment]
This PR is really simple, PTAL @lzmhhh123 @eurekaka @winoros

File: planner/optimize.go [hunk 1]
```diff
@@ -214,6 +214,9 @@ func getBindRecord(ctx sessionctx.Context, stmt ast.StmtNode) (*bindinfo.BindRec
 		return nil, ""
 	}
 	globalHandle := domain.GetDomain(ctx).BindHandle()
+	if globalHandle == nil {
```
Comment (eurekaka) [inline planner/optimize.go]: When would this happen?

Comment (qw4990) [reply planner/optimize.go]: When TiDB starts, we may use the domain's handler(in `LoadPrivilegeLoop`) before initializing it (in `LoadBindLoop`).
image

Comment (qw4990) [reply planner/optimize.go]: PTAL @eurekaka

**Pull request merged**